shim_maker.py

```diff
@@ -21,62 +26,138 @@ def fancy_list(prefix: str, to_join: Iterable[str], limit: int = 100, with_brack
-            with open(f"{shim_path}/{fn}", "w", encoding="utf-8") as f:
-                f.write(content)
-
-        elif fn.startswith("__"):
-            pass
+        code += "\n\n" + new_all
+
+    code += f"\n\n# isort: split\nfrom {module_name} import __dict__ as __original_dict__\nlocals().update(__original_dict__)"
+
+    return docstring, sort_imports(code)
+
+
+def shim_folder(path: pathlib.Path, pypath: str, shim_path: pathlib.Path) -> List[pathlib.Path]:
+    modified = []
+    for fn in path.iterdir():
+        mod = fn.relative_to(path)
+
+        if not (shim_path / mod).parent.exists():
+            (shim_path / mod).parent.mkdir(exist_ok=True, parents=True)
+            print("Created directory: ", (shim_path / mod))
+
+        if fn.name in ("__init__.py", "__main__.py", "py.typed") and "tasks" != fn.parent.name:
+            with open(path / mod, "r", encoding="utf-8") as fr:
+                try:
+                    with open(shim_path / mod, "r", encoding="utf-8") as fw:
+                        if fw.read() == fr.read():
+                            continue
+                except FileNotFoundError:
+                    pass
+                fr.seek(0)
+                print("Updating file: ", shim_path / mod)
+                with open(shim_path / mod, "w", encoding="utf-8") as fw:
+                    fw.write(fr.read())
+            modified.append(shim_path / mod)
+
+        elif fn.suffix == ".py":
+            docstring, imports = create_file(f"{pypath}.{mod.stem}")
+            to_write = ""
+            if docstring:
+                to_write += '"""\n' + docstring.strip() + '\n"""\n\n'
+            to_write += imports
+            try:
+                with open(shim_path / mod) as f:
+                    existing = f.read()
+            except FileNotFoundError:
+                existing = None
+            if existing != to_write:
+                print("Updating file: ", shim_path / mod)
+                with open(shim_path / mod, "w", encoding="utf-8") as f:
+                    f.write(to_write)
+
+                modified.append(shim_path / mod)
+
+        elif fn.is_dir():
+            (shim_path / mod).mkdir(parents=True, exist_ok=True)
+            modified.extend(shim_folder(fn, f"{pypath}.{fn.stem}", shim_path / mod))
+
+    return modified
+
+
+def shim(path: pathlib.Path, shim_path: pathlib.Path) -> None:
+    shim_path.mkdir(parents=True, exist_ok=True)
+    return shim_folder(path, path.stem, shim_path)
+
+
+def main():
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("path", help="Path to the disnake module.")
+    parser.add_argument(
+        "-s",
+        "--shim",
+        help="Name of the shim module.",
+        default="discord",
+    )
+    args = parser.parse_args()
+    try:
+        path = pathlib.Path(args.path)
+    except ValueError:
+        print(f"Invalid path to package to be shimmed: {args.path}")
+        return 2
```

Comment:
```suggestion
        print(f"Invalid path to package to be shimmed: {args.path}", file=sys.stderr)
        sys.exit(2)
```

Reply:
Won't implement this, the main method shouldn't exit.

Reply:
The reason for doing this was that `len(res)` below could otherwise fail, since `res` is `List[Path] | Literal[2, 1, 0]`